ewatercycle/models/marrmot.py

```diff
@@ -38,7 +38,7 @@ def _generate_cfg_dir(cfg_dir: Path = None) -> Path:
         scratch_dir = CFG['output_dir']
         # TODO this timestamp isnot safe for parallel processing
         timestamp = datetime.datetime.now(datetime.timezone.utc).strftime("%Y%m%d_%H%M%S")
-        cfg_dir = Path(scratch_dir) / f'marrmot_{timestamp}'
+        cfg_dir = to_absolute_path(f'marrmot_{timestamp}', parent=Path(scratch_dir))
     cfg_dir.mkdir(parents=True, exist_ok=True)
     return cfg_dir
 
```

Comment:
On line 117 and 361 we could also use `to_absolute_path(cfg_dir)` instead of `Path(cfg_dir)`

Reply:
Even `must_exist=True` could be set.